gateway/api/views.py

```diff
@@ -325,6 +297,68 @@ def run(self, request):
 
         return Response(job_serializer.data)
 
+    @action(methods=["GET"], detail=False, url_path="get_by_title/(?P<title>[^/.]+)")
+    def get_by_title(self, request, title):
```

Comment:
Sorry Iskandar, this is new for me. What are we doing with that `re`?

Reply:
in order to pass parameter `title` in a form of url `v1/api/programs/get_by_title/<title>` we need to add ths `url_path` parameter.

Reply:
it is basically defining path by some id

Reply:
if this id is not `primary key` and title is not a primary key

Reply:
ok, it's not like we are limitating the usage of spaces or something like that, just to verify that we are not changing this "Docker-Image-Program" due to that.

Reply:
yes, for spaces one just need to encode it to url str

On that note. We probably need to sanitize titles and do not allow spaces and special chars

Reply:
> On that note. We probably need to sanitize titles and do not allow spaces and special chars

Yes, totally agree on that.